docs/src/rules/no-useless-escape.md

```diff
@@ -67,6 +67,42 @@ Examples of **correct** code for this rule:
 
 :::
 
+## Options
+
+This rule has an object option:
+
+* `allowRegexCharacters` - An array of characters that should be allowed to have unnecessary escapes in regular expressions. This is useful for characters like `-` where escaping can prevent accidental character ranges. For example, in `/[0\-]/`, the escape is technically unnecessary but helps prevent the pattern from becoming a range if another character is added later (e.g., `/[0\-9]/` vs `/[0-9]/`).
+
+### allowRegexCharacters
+
+Examples of **incorrect** code for the `{ "allowRegexCharacters": ["-"] }` option:
+
+::: incorrect
+
+```js
+/*eslint no-useless-escape: ["error", { "allowRegexCharacters": ["-"] }]*/
+
+/\!/;
+/\@/;
+/[a-z\^]/;
+```
+
+:::
+
+Examples of **correct** code for the `{ "allowRegexCharacters": ["-"] }` option:
+
+::: correct
+
+```js
+/*eslint no-useless-escape: ["error", { "allowRegexCharacters": ["-"] }]*/
+
+/[0\-]/;
+/[a\-z]/;
+/[0\-9]/;
```

Comment:
These two examples could be misleading because these are not useless escapes so they wouldn't be reported even without the option. Can we replace them with something else, for example:

```suggestion
/[\-9]/;
/a\-b/;
```